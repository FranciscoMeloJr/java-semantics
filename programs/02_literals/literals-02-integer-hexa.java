//Hexadecimal integer literals
public class main {
	static void main(String[] args) {
    System.out.println("Hexa int literals:");
		System.out.println(0x0 + " " + 0x5f + " " + 0X5F + " " + 0xFFFFFFF0);
    System.out.print("Hexa int overflow, 0x70000000 + 0x70000000 = ");
		System.out.println(0x70000000 + 0x70000000);
    System.out.println("Hexa long literals:");
		System.out.println(0x0L + " " + 0x123456789abL + " " + 0x123456789ABl
        + " " + 0xFFFFFFF0L + " " + 0xFFFFFFFF00000000L);
    System.out.print("Hexa long not overflow, 0x70000000L + 0x70000000L = ");
		System.out.println(0x70000000L + 0x70000000L);
		System.out.println("Done!");
	}
}

