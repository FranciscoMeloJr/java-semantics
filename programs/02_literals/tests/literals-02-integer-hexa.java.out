Hexa int literals:
0 95 95 -16
Hexa int overflow, 0x70000000 + 0x70000000 = -536870912
Hexa long literals:
0 1250999896491 1250999896491 4294967280 -4294967296
Hexa long not overflow, 0x70000000L + 0x70000000L = 3758096384
Done!
